Fluid/Filters/MiscFilters.cs

```diff
@@ -614,14 +614,24 @@ private static async ValueTask WriteJson(Utf8JsonWriter writer, FluidValue input
                                 value = access.Get(obj, name, ctx);
                             }
 
+                            if (value is object)
+                            {
+                                stack ??= new HashSet<object>();
+                                if (stack.Contains(value))
+                                {
+                                    value = "circular reference detected.";
+                                }
+                            }
+
                             var fluidValue = FluidValue.Create(value, ctx.Options);
                             if (fluidValue.IsNil())
                             {
                                 continue;
                             }
 
                             writer.WritePropertyName(name);
-                            await WriteJson(writer, fluidValue, ctx);
+                            stack.Add(obj);
+                            await WriteJson(writer, fluidValue, ctx, new HashSet<object>(stack));
```

Comment:
I think you can pass the same `stack` instance and remove the object from it right after the call. This will save some allocations.

Reply:
Let me give that a whirl.  I just wrote another test that covers the sibling path as well.

Reply:
@sebastienros I added that optimization, good call!  Tests pass, including the new one I added to make sure the recursion stack doesn't bleed down a sibling property path.